sql/sql_show.cc

```diff
@@ -5882,7 +5882,7 @@ static int get_schema_views_record(THD *thd, TABLE_LIST *tables,
         {
           TABLE_LIST table_list;
           uint view_access;
-          memset(&table_list, 0, sizeof(table_list));
+          memset(static_cast<void*>(&table_list), 0, sizeof(table_list));
```

Comment:
This does not fix the root cause of the warning. If the actual fix is too complex for us at this point of 5.5 lifecycle, just patch the build for -Wno-class-memaccess instead.